deployment/windows/jellyfin.nsi

```diff
@@ -0,0 +1,202 @@
+    CreateShortCut "$DESKTOP\Jellyfin.lnk" "$INSTDIR\jellyfin.exe"
+SectionEnd
+
+Section /o "Jellyfin Service" InstallService
+    ExecWait '"$INSTDIR\nssm.exe" install Jellyfin "$INSTDIR\jellyfin.exe" --datadir "$_JELLYFINDATADIR_"' $0
+    DetailPrint "Jellyfin Service install, $0"
+    Sleep 3000
+    ExecWait '"$INSTDIR\nssm.exe" set Jellyfin Start SERVICE_DELAYED_AUTO_START' $0
+    DetailPrint "Jellyfin Service setting, $0"
+    StrCpy $_SERVICEINSTALLED_ "YES"
+SectionEnd
+
+Section /o "Start Jellyfin after installation" LaunchJellyfin
+; either start the service or launch jellyfin standalone
+    StrCmp $_SERVICEINSTALLED_ "YES" ServiceStart Standalone
+    
+    ServiceStart:
+    ExecWait '"$INSTDIR\nssm.exe" start Jellyfin' $0
+    DetailPrint "Jellyfin service start, $0"
+    Return
+    
+    Standalone:
+    ExecWait '"$INSTDIR"\jellyfin.exe' $0
+    DetailPrint "$INSTDIR\jellyfin.exe start, $0"
+
+SectionEnd
+
+; This can be uncommented in case Emby Migration is planned later
+;Section /o "Migrate Emby Library" MigrateEmbyLibrary
+;    DetailPrint "Migrating Emby Library"
+;    CopyFiles $_DEFAULTEMBYDATADIR_/config $_JELLYFINDATADIR_
+;    CopyFiles $_DEFAULTEMBYDATADIR_/cache $_JELLYFINDATADIR_
+;    CopyFiles $_DEFAULTEMBYDATADIR_/data $_JELLYFINDATADIR_
+;    CopyFiles $_DEFAULTEMBYDATADIR_/metadata $_JELLYFINDATADIR_
+;    CopyFiles $_DEFAULTEMBYDATADIR_/root $_JELLYFINDATADIR_
+
+;SectionEnd
+
+
+;--------------------------------
+;Descriptions
+
+  ;Language strings
+    LangString DESC_InstallJellyfin ${LANG_ENGLISH} "Install Jellyfin"
+    LangString DESC_InstallService ${LANG_ENGLISH} "Install As a Service"
+    LangString DESC_DesktopShortcut ${LANG_ENGLISH} "Create a desktop shortcut"
+    LangString DESC_LaunchJellyfin ${LANG_ENGLISH} "Start Jellyfin after Install"
+;    LangString DESC_MigrateEmbyLibrary ${LANG_ENGLISH} "Migrate existing Emby Library"
+
+  ;Assign language strings to sections
+    !insertmacro MUI_FUNCTION_DESCRIPTION_BEGIN
+    !insertmacro MUI_DESCRIPTION_TEXT ${InstallJellyfin} $(DESC_InstallJellyfin)
+    !insertmacro MUI_DESCRIPTION_TEXT ${DesktopShortcut} $(DESC_DesktopShortcut)
+    !insertmacro MUI_DESCRIPTION_TEXT ${InstallService} $(DESC_InstallService)
+    !insertmacro MUI_DESCRIPTION_TEXT ${LaunchJellyfin} $(DESC_LaunchJellyfin)
+;    !insertmacro MUI_DESCRIPTION_TEXT ${MigrateEmbyLibrary} $(DESC_MigrateEmbyLibrary)
+    !insertmacro MUI_FUNCTION_DESCRIPTION_END
+
+;--------------------------------
+;Uninstaller Section
+
+Section "Uninstall"
+    SetShellVarContext current
+    StrCpy $_JELLYFINDATADIR_ "$LOCALAPPDATA\jellyfin\"
+;TODO
+; stop running instance
+    ExecWait '"$INSTDIR\nssm.exe" stop Jellyfin' $0
+    DetailPrint "Jellyfin service stop, $0"
+    ExecWait '"$INSTDIR\nssm.exe" remove Jellyfin confirm' $0
+    DetailPrint "Jellyfin Service remove, $0"
+
+    Delete "$INSTDIR\Uninstall.exe"
+    RMDir /r "$INSTDIR"
+    RMDir /r "$_JELLYFINDATADIR_"
```

Comment:
The option to keep data when uninstalling should be present.

Reply:
Agree, done.